pkg/subroutines/creator.go

```diff
@@ -0,0 +1,97 @@
+package subroutines
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openmfp/golang-commons/controller/lifecycle"
+	"github.com/openmfp/golang-commons/errors"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1alpha1 "github.com/openmfp/account-operator/api/v1alpha1"
+)
+
+type CreatorSubroutine struct {
+	client  openfgav1.OpenFGAServiceClient
+	storeID string
+}
+
+func NewCreatorSubroutine(fgaGrpcAddr, storeId string) *CreatorSubroutine {
+	conn, _ := grpc.NewClient(fgaGrpcAddr,
+		grpc.EmptyDialOption{},
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
+	)
+
+	cl := openfgav1.NewOpenFGAServiceClient(conn)
+
+	return &CreatorSubroutine{client: cl, storeID: storeId}
+}
+
+func (e *CreatorSubroutine) Process(ctx context.Context, runtimeObj lifecycle.RuntimeObject) (ctrl.Result, errors.OperatorError) {
+	account := runtimeObj.(*v1alpha1.Account)
+
+	for _, condition := range account.Status.Conditions {
+		if condition.Reason == "AccountReady" && condition.Type == "Ready" && condition.Status == "True" {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	_, err := e.client.Write(ctx, &openfgav1.WriteRequest{
+		StoreId: e.storeID,
+		Writes: &openfgav1.WriteRequestWrites{
+			TupleKeys: []*openfgav1.TupleKey{
+				{
+					Object:   fmt.Sprintf("account:%s", account.Name),
+					Relation: "owner",
+					User:     fmt.Sprintf("user:%s", *account.Spec.Creator),
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		return ctrl.Result{}, errors.NewOperatorError(err, true, true)
+	}
+
+	account.Status.Conditions = append(account.Status.Conditions, metav1.Condition{
+		Type:    "Ready",
+		Status:  "True",
+		Reason:  "AccountReady",
+		Message: "Message is ready",
+	})
+
+	return ctrl.Result{}, nil
+}
+
+func (e *CreatorSubroutine) Finalize(ctx context.Context, runtimeObj lifecycle.RuntimeObject) (ctrl.Result, errors.OperatorError) {
+	account := runtimeObj.(*v1alpha1.Account)
+
```

Comment:
As already mentioned above, also the finalization logic needs to contain a check for an emtpy `account.Spec.Creator` in order to prevent unintended crashes